csi.proto

```diff
@@ -1295,6 +1306,20 @@ message NodePublishVolumeRequest {
   // This field is OPTIONAL and MUST match the volume_context of the
   // volume identified by `volume_id`.
   map<string, string> volume_context = 8;
+
+  // If SP has VOLUME_MOUNT_GROUP node capability and CO provides
+  // this field then SP MUST ensure that volume is mounted with
+  // provided volume_mount_group and all files and directories
+  // within the volume are readable and writable by the provided
+  // volume_mount_group.
+  // If NodeStageVolume was previously called with volume_mount_group
+  // CO MUST ensure that NodePublishVolume uses the same
```

Comment:
I thought we said here this is a SHOULD not a MUST. I think the semantics we want is that, if the CO does pass the same volume_mount_group as it did to stage, then the SP MUST NOT fail for reason related to the group. If the CO passes a different group, then the operation MAY fail and that should be communicated in the error code.

Reply:
I thought - if `NodePublish` gets called with different value than `NodeStage` - it just isn't going to work for ANY storage provider because volume was already staged with a different group identifier.

The place where we wanted to use SHOULD or MAY is - if SP did not had `NodeStage` but two or more `NodePublish` was called with different `volume_mount_group`. Such a case MAY or MAY not be supported by SP and hence I covered this case with an error code:

> | Volume already published with different volume_mount_group | 9 FAILED_PRECONDITION | Indicates that the volume with specified `volume_id` was already node published using different `volume_mount_group` on this node and can not be node published. | Caller MUST ensure that `NodePublishVolume` is called with same `volume_mount_group` on all target paths. |